packages/fetchai/contracts/oracle/contract.py

```diff
@@ -0,0 +1,130 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains the class to connect to an Oracle contract."""
+
+import logging
+from typing import Any, Dict
+
+from vyper.utils import keccak256
+
+from aea.common import Address
+from aea.configurations.base import PublicId
+from aea.contracts.base import Contract
+from aea.crypto.base import LedgerApi
+from aea.crypto.ethereum import EthereumApi
+
+
+PUBLIC_ID = PublicId.from_str("fetchai/oracle:0.1.0")
+CONTRACT_ROLE = keccak256(b"ORACLE_ROLE")
+
+_default_logger = logging.getLogger("aea.packages.fetchai.contracts.oracle.contract")
+
+
+class FetchOracleContract(Contract):
+    """The Fetch oracle contract."""
+
+    @classmethod
+    def get_grant_role_transaction(
+        cls,
+        ledger_api: LedgerApi,
+        contract_address: Address,
+        oracle_address: Address,
+        gas: int = 30000,
```

Comment:
might be good to put this in a top-module variable

Reply:
Or maybe we should set the default limit to zero to make them fail if no gas limit is explicitly set? Thoughts @marcofavorito @DavidMinarsch ?

Reply:
It's not a bad idea. One advantage is that things will not work "by chance" if you forget to pass the `gas` parameter.

Reply:
Yes, that's what I was thinking too.